### ISSUE TITLE
Unexpected W0127 when assigning to a class variable

### ISSUE THREAD

**Issue by OddBloke** [PyCQA/pylint#3144]
### Steps to reproduce
1. Write out this content to `reproducer.py`:

```py
schema = "test"


class MyClass:
    schema = schema

    def foo(self):
        print(self.schema)


MyClass().foo()
```

2. Run `pylint reproducer.py`

### Current behavior

```
reproducer.py:5: [W0127(self-assigning-variable), MyClass] Assigning the same variable 'schema' to itself
```

### Expected behavior

No warning, as the class variable is not the same as the module-level variable.

### pylint --version output

```
pylint 2.4.1
astroid 2.3.0
Python 3.7.2+ (default, Feb  2 2019, 14:31:48) 
[GCC 8.2.0]
```

**Comment by YakDriver** [comment]
Is the point that it would be clearer to do this instead?

```python
schema = "test"

class MyClass:
    MyClass.schema = schema

    def foo(self):
        print(self.schema)
```

**Comment by paride** [comment]
I don't think that's going to work (`MyClass` not defined within the `MyClass` class).

**Comment by PCManticore** [comment]
This is already fixed in `master` by https://github.com/PyCQA/pylint/commit/d8fae0124b60abb40d833e78fb32fe355143889b which will be part of `pylint 2.4.2` shortly (this weekend)
